### DISCUSSION TITLE
Just curious why not use useSyncExternalStore like Zustand?

### DISCUSSION THREAD

**Discussion by L-Qun** [pmndrs/jotai#2137] [Q&A]
https://codesandbox.io/s/react-tearing-jotai-wqwwqt?file=/src/Controller.js
![20230917132724_rec_](https://github.com/pmndrs/jotai/assets/68707557/05b67673-a83b-49df-8827-625074e1db6d)
You can see that there are tearing issues between the individual components, and React18 also remove eagerState for useReducer.

**Comment by dai-shi** [comment]
uSES fixes tearing but doesn't behave nicely with promises which is one of the goals of Jotai.
I made non-uSES version of zustand too: https://github.com/dai-shi/use-zustand

**Reply by L-Qun** [reply]
I don't understand what you mean. Why it doesn't behave nicely with promises? Doesn't useReducer also need **use** hooks to deal with promise
![image](https://github.com/pmndrs/jotai/assets/68707557/5c9c1a0c-268c-4ccf-abec-f468327cc461)

**Reply by dai-shi** [reply]
Sorry for the confusion. It's about the behavior with useTransition. There's a Twitter thread when we discussed about it and developed `use-zustand`. I'll look for it. Or, maybe I can write a blog post.

**Reply by L-Qun** [reply]
Thank you for your reply, I will wait for your blog and continue to understand it

**Reply by iwoplaza** [reply]
I would love a blog post on the behavior of `useTransition` and `Jotai`, or other state management libraries. I tried incorporating `useTransition` into our app using Jotai, but I have not managed to figure out how to make that work.

Does using `useTransition` alongside `useSyncExternalStore` cause some stale-UI or invalid state, or is it just harder to wrap state updates in a transition while using `uSES`?

**Reply by dai-shi** [reply]
If you use `useSyncExternalStore`, basically time slicing doesn't work.
https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-rendering
☝️ Compare `jotai` and `zustand`

See also: https://github.com/facebook/react/issues/26814

**Reply by L-Qun** [reply]
So the reason you don't choose useSyncExternalStore is because it doesn't support time slicing

**Reply by dai-shi** [reply]
Yeah, that what I meant with "behave nicely with promises" for useTransition.

**Reply by dai-shi** [reply]
https://blog.axlight.com/posts/why-use-sync-external-store-is-not-used-in-jotai/ I'm basically rephrasing what I said above.

**Comment by L-Qun** [comment]
@dai-shi And another question is why you use useReducer instead of useState. I think the performance of useState is better than useReducer because the React team has already removed eager state policy from useReducer: https://github.com/facebook/react/pull/22445

**Reply by L-Qun** [reply]
Looking forward to your reply❤️

**Reply by dai-shi** [reply]
I think we actually depend on the lazy updates. Or what I call the cheat mode. There must be some tests failing if we change  useReducer to useState.
